torch/_dynamo/variables/builder.py

```diff
@@ -706,7 +716,18 @@ def wrap_literal(self, value):
             )
 
     def wrap_tensor(self, value: torch.Tensor):
-        if self.get_source().guard_source().is_nn_module():
+        source = self.get_source()
+        # TODO: This is a *hack*. We have `FSDPNNModuleSource.guard_source()`
+        # emulate `NNModuleSource.guard_source()` so that `is_nn_module()`
+        # returns true, skipping guard creation. However, for FSDP, we *do not*
+        # want to register the "parameters" (which are `Tensor` views) to the
+        # fake root module since they are transient. We need a way to make this
+        # distinction. For now, we hard code against `AttrSource`.
+        skip_register_attr = isinstance(source, FSDPNNModuleSource) or (
+            isinstance(source, AttrSource)
+            and isinstance(source.base, FSDPNNModuleSource)
+        )
+        if source.guard_source().is_nn_module() and not skip_register_attr:
```

Comment:
hmm, is the 'hacky' part that you have to check recursively for FSDPNNModuleSource in this code?

how does guard_source().is_nn_module() work?  is that implementing the recursive part?

if so, maybe just add `guard_source().is_fsdp_module()` and its good to go?